### PR TITLE
Add Initial Support for Python 3.7 and 3.8, Remove Support for 3.4

### PR DESCRIPTION
**Pull request by danyeaw**
Adds initial support for Python 3.7 and 3.8:
- Adds magic numbers for 3.7 and 3.8
- Add support for PEP 552 which uses 4 32-bit words in the pyc header in Python 3.7 and up
- Adds all opcodes in the virtual machine, special support for the opcodes has not been added yet
- Update the docs to have users use Python 3.5 and 3.6, and Node 10.x LTR

- The testserver runs with Python 3.7/3.8 and Node 10.x
- Some tests pass, but support for some of the new opcodes like LOAD_METHOD will come in follow on PRs.

Thanks to @freakboy3742 and @dgelessus for their help and support on gitter as I start to get my feet wet :heart:

## PR Checklist:

- [ ] All new features have been tested
- [ ] All new features have been documented
- [X] I have read the **CONTRIBUTING.md** file
- [X] I will abide by the code of conduct

### CODE REVIEW

File: batavia/modules/dis.js [hunk 1]
```diff
@@ -84,6 +87,10 @@ def_unary_op('UNARY_NOT', 12)
 
 def_unary_op('UNARY_INVERT', 15)
 
+// Introduced in Python 3.5
+// def_binary_op('BINARY_MATRIX_MULTIPLY', 16)
+// def_binary_op('INPLACE_MATRIX_MULTIPLY', 17)
```
[the remark this answers is not in the document]
Comment (danyeaw) [reply batavia/modules/dis.js]: Thanks for catching this, this was commented out when I was troubleshooting op code type 16 errors, and I missed restoring it.

File: beekeeper.yml [hunk 2]
```diff
@@ -9,22 +9,22 @@ pull_request:
...
-                name: Python 3.5 tests
-                task: batavia-py35
-                critical: False
             - py3.6:
                 name: Python 3.6 tests
                 task: batavia-py36
                 critical: False
+            - py3.7:
+                name: Python 3.7 tests
+                task: batavia-py37
```
[the remark this answers is not in the document]
Comment (danyeaw) [reply beekeeper.yml]: Yup :+1:

File: docs/how-to/contribute-code.rst [hunk 3]
```diff
@@ -52,7 +52,7 @@ Install Node.JS
 ^^^^^^^^^^^^^^^
 
 Lastly, you'll need to install `Node.js`_. You need to have a recent version
-of Node; we test using v6.9.1. Once you've installed node, you can use it to
+of Node; we test using v10.x. Once you've installed node, you can use it to
```
[the remark this answers is not in the document]
Comment (danyeaw) [reply docs/how-to/contribute-code.rst]: :+1:

File: batavia/core/version.js [hunk 4]
```diff
@@ -8,7 +8,10 @@ magic_map[String.fromCharCode(238, 12, 13, 10)] = [[3, 4], [0xF]]
 magic_map[String.fromCharCode(248, 12, 13, 10)] = [[3, 5], [0xA, 0]]
 magic_map[String.fromCharCode(22, 13, 13, 10)] = [[3, 5], [0xF]]
 magic_map[String.fromCharCode(23, 13, 13, 10)] = [[3, 5, 3], [0xF]]
-magic_map[String.fromCharCode(51, 13, 13, 10)] = [[3, 6, 2], [0xF]]
+magic_map[String.fromCharCode(23, 13, 13, 10)] = [[3, 5], [0xF]]
```
Comment (martica) [inline batavia/core/version.js]: This looks suspicious to me. It's overwriting the value from the previous line, isn't it?

Comment (danyeaw) [reply batavia/core/version.js]: @martica Thanks, I cleaned this duplicate up and removed all the version that are < 3.5 :+1:

File: tests/utils.py [hunk 5]
```diff
@@ -1143,6 +1137,9 @@ def assertBinaryOperation(self, x_values, y_values, operation, format, substitut
         vars()['test_ne_%s' % datatype] = _binary_test(
             'test_ne_%s' % datatype, 'x != y', examples
         )
+        vars()['test_matmul_%s' % datatype] = _binary_test(
+            'test_matmul_%s' % datatype, 'x * y', examples
```
Comment (martica) [inline tests/utils.py]: This and the inplace one below should use @ not *.

Comment (danyeaw) [reply tests/utils.py]: Updated to use small ints :+1:

**Comment by danyeaw** [comment]
> version.later and version.earlier are confusing.
> 
> This is doing version is later than 3.5.nothing, so all 3.5.0 and forward are later than that.

Thanks forgot about that :+1:

File: batavia/core/types/PYCFile.js [hunk 6]
```diff
@@ -11,17 +12,27 @@ var PYCFile = function(data) {
...
+    this.magic = data.slice(0, 4);
     constants.BATAVIA_MAGIC = String.fromCharCode(
         this.magic[0],
         this.magic[1],
         this.magic[2],
         this.magic[3]
     )
+    
+    // Support PEP 552 for 4 32-bt words in Python 3.7
+    if (version.later('3.7')) {
```
Comment (martica) [inline batavia/core/types/PYCFile.js]: Sorry. I mixed this up. Should be 3.6 here.

**Pull request closed without merging**